content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md

```diff
@@ -0,0 +1,113 @@
+	<center><figcaption>Selecting Directories in the New Find Panel (contd.)</figcaption></center>
+</figure>
+
+<figure>
+	<img src="/files/blog/calisto-mathias/Final-Report/directories-entries-in-menu.png" alt="New Find Panel Directory Selector" width="598px">
+	<center><figcaption>Selecting Directories in the New Find Panel (contd.)</figcaption></center>
+</figure>
+
+### Adding the Design for an Incremental Search Find Panel
+
+The most crucial and challenging aspect of my project was implementing the design for the incremental search find panel, which associates attribute columns with their corresponding search columns. While I managed to implement the core functionality, there are a few limitations that need to be addressed in the future.
+
+Currently, the incremental find panel integrated into the Find Panel results window is under review on GitHub (and soon on Gerrit). It will take some time before it is merged into the main codebase. Despite the challenges, I managed to get the core functionality working, allowing users to pause and restart their queries, as well as adjust search terms, all within the results window without switching back and forth repeatedly.
+
+<img src="/files/blog/calisto-mathias/Final-Report/incremental-search-first-look.png" alt="Incremental Search panel">
+<br><br>
+
+The user can select the various methods of combination in the same column. This combination follows the same rules as the searcy "by-attribute" mode in the old find panel. 
+
+<img src="/files/blog/calisto-mathias/Final-Report/selecting-combination-options.png" alt="Incremental Search Panel">
+<br><br>
+
+Apart from this, users can pause and re-execute their queries on the file-system without any hassle now. The columns can also be moved around with each other, in order to set more powerful combination modes. The only caveat at the moment is that the two different combination modes for the incremental find panel hasn't been completed. As of now, the columns are always combined using the "AND" operator (so the order does not matter, but this is temporary!)
+
+## What’s Still Left to Do?
+
+Implementing this project in one go was quite challenging. Initially, I hoped to complete everything by the project deadline, but that proved unrealistic. It took me some time to grasp the technical details involved. However, I believe I can complete the remaining work with a bit more time and by tackling it in smaller steps.
+
+Here’s what remains to be done in the incremental find panel:
+
+1. **Incremental Search with Keystrokes:** The goal was to refresh results with each keystroke. Although I implemented this, it turned out to be unstable due to the high load from frequent clearing, refreshing, and restarting of queries. I had planned to develop a debouncer to manage this load but didn’t manage to complete it. This will be my primary focus after the GSoC’24 period ends.
```

Comment:
Maybe expand this section a little bit on which strategies you would like to explore, so that in case you are not able to finish it due to other obligations, others will have a head start on doing this.

Reply:
Yup. I wrote down what I was thinking of doing. Thanks :)